pynest/nest/tests/test_kinetics_multi_cmpt_neu.py

```diff
@@ -0,0 +1,367 @@
+
+		
+		return f
+	
+
+		
+
+	def __init__(self,active,resolution):
+		
+		self.num_cmpt = 3
+		self.cmpt_cnt = 0
+		self.active_flag = active
+		self.resolution = resolution
+		self.V_th = -45.0         		   #mV
+		self.dist_V_th = -20.0
+		self.V_T = self.V_th
+		self.dist_VT = self.dist_V_th
+		self.V_reset = -60.0                    #mV
+	    	self.T_ref = 2.0                        #ms
+		self.G_conn = [25.,25.]		   #nS
+
+
+		self.tau_VT = 7.
+		self.VT_jump = 25.
+		self.reset_time = int(self.T_ref/self.resolution)
+		self.tau_distVT = 20.
+		self.dist_VT_jump = 20.
+		self.counter = 0
+		self.ref = 0
+	    	self.E_l = np.array([[-70.0],[-65.0],[-60.0]])          #mV
+		self.Conn_mat = np.array([[self.G_conn[0],-self.G_conn[0],0.],[-self.G_conn[0],self.G_conn[0]+self.G_conn[1]+self.E_l[1],-self.G_conn[1]],[0.,-self.G_conn[1],self.G_conn[1]+self.E_l[2]]])
+		self.G_l = np.array([[5.0],[5.0],[5.0]])                #nS
+	    	self.C_m = np.array([[43.0],[18.0],[13.0]])           #pF
+
+	    	self.E_ex = np.array([[0.0],[0.0],[0.0]])               #mV
+	    	self.E_in = np.array([[-85.0],[-85.0],[-85.0]])         #mV
+
+	    	self.Tau_synE = np.array([[1.0],[1.0],[1.0]])           #ms
+	    	self.Tau_synI = np.array([[2.0],[2.0],[2.0]])           #ms
+		self.u = np.array([-70.0,-65.0,-60.0,0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,0.,0.]) 
+		self.G_exc = np.array([[0.0],[0.0],[0.0]])
+		self.G_inh = np.array([[0.0],[0.0],[0.0]])
+		self.G_ca = np.array([[0.0],[0.0],[0.0]])
+		self.G_exc1 = np.array([[0.0],[0.0],[0.0]])
+		self.G_inh1 = np.array([[0.0],[0.0],[0.0]])
+		self.G_ca1 = np.array([[0.0],[0.0],[0.0]])
+		self.I_conn = 0.0
+		self.I_curr = np.array([[0.0],[0.0],[0.0]])
+
+		self.y_exc = np.array([[[0.],[0.]],[[0.],[0.]],[[0.],[0.]]])
+		self.y_inh = np.array([[[0.],[0.]],[[0.],[0.]],[[0.],[0.]]])
+		self.y_ca  = np.array([[[0.],[0.]],[[0.],[0.]],[[0.],[0.]]])
+
+		# distal_spike parameters
+
+		self.eca    = 50.0 	
+		self.gca    = 70.0    
+		self.taum   = 15.0    
+		self.tauh   = 80.0    
+		self.halfm  = -9.0    
+		self.halfh  = -21.0   
+		self.slopem = 0.5     
+		self.slopeh = -0.5  
+
+		self.u[15] = 1.0/(1.0+np.exp((self.u[2]-self.halfm)*-1.0*self.slopem))
+		self.u[16] = 1.0/(1.0+np.exp((self.u[2]-self.halfh)*-1.0*self.slopeh))
+
+
+		self.r = ode(self.fun_u).set_integrator('dop853',rtol=1e-3, nsteps=500)
+		self.r.set_initial_value(self.u,0.)
+	
+	def compute_u(self,G_exc,G_inh,I_curr):
+
+
+
+		self.counter = 0
+		self.distal_cnt = 0
+
+		self.r.y[11] += (G_exc[0,0]*np.exp(1)/self.Tau_synE[0,0])
+		self.r.y[13] += (G_inh[0,0]*np.exp(1)/self.Tau_synI[0,0])
+		self.r.y[7] += (G_exc[1,0]*np.exp(1)/self.Tau_synE[1,0])
+		self.r.y[9] += (G_inh[1,0]*np.exp(1)/self.Tau_synI[1,0])
+		self.r.y[3] += (G_exc[2,0]*np.exp(1)/self.Tau_synE[2,0])
+		self.r.y[5] += (G_inh[2,0]*np.exp(1)/self.Tau_synI[2,0])
+
+		self.r.integrate(self.r.t+self.resolution)
+		self.u = self.r.y
+
+
+class KineticsMultiCmptNeuTestCase(unittest.TestCase):		
```

Comment:
Hi Tammo,

the error persists even after making the corrections

yansong

On Tue, Dec 8, 2015 at 2:13 PM, Tammo Ippen <notifications@github.com>
wrote:

> In pynest/nest/tests/test_kinetics_multi_cmpt_neu.py
> <https://github.com/nest/nest-simulator/pull/176#discussion_r46950661>:
>
> > +
> > +		self.counter = 0
> > +		self.distal_cnt = 0
> > +
> > +		self.r.y[11] += (G_exc[0,0]*np.exp(1)/self.Tau_synE[0,0])
> > +		self.r.y[13] += (G_inh[0,0]*np.exp(1)/self.Tau_synI[0,0])
> > +		self.r.y[7] += (G_exc[1,0]*np.exp(1)/self.Tau_synE[1,0])
> > +		self.r.y[9] += (G_inh[1,0]*np.exp(1)/self.Tau_synI[1,0])
> > +		self.r.y[3] += (G_exc[2,0]*np.exp(1)/self.Tau_synE[2,0])
> > +		self.r.y[5] += (G_inh[2,0]*np.exp(1)/self.Tau_synI[2,0])
> > +
> > +		self.r.integrate(self.r.t+self.resolution)
> > +		self.u = self.r.y
> > +
> > +
> > +class KineticsMultiCmptNeuTestCase(unittest.TestCase):		
>
> This TestCase also needs the decorators:
>
> @nest.check_stack
> @unittest.skipIf(not HAVE_GSL, 'GSL is not available')
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nest/nest-simulator/pull/176/files#r46950661>.
>